### ISSUE TITLE
why Avalanch takes longer time per epoch for a subset of the dataset (experience) ?

### ISSUE THREAD

**Issue by 3koozy** [ContinualAI/avalanche#1074]
🐛 **Describe the bug**
why continual training takes a long time per epoch for a subset of the dataset (experience) ?
I am fine-tuning an EffcientNet_b4 model on my dataset using joint training (without avalanche) and it takes less than 20 seconds
to go over the entire dataset (UC Merced) per epoch.

but when I use a new class scenario with the naive strategy it takes over 1 min to finish 1 epoch of training of a subset of the dataset (7 experiences = 1/7).

Also tested with different strategies with the same result (replay).

🐜 **To Reproduce**
A [minimal working example](https://en.wikipedia.org/wiki/Minimal_working_example) to reproduce the issue. The code should be executable without modifications.

🐝 **Expected behavior**

🐞 **Screenshots**

🦋 **Additional context**
Add any other context about the problem here like your python setup.

**Comment by HamedHemati** [comment]
Hi,

It makes sense that Avalanche is slower than a regular training code. The main reason is the template-based training with predefined callbacks that Avalanche uses. This means you have many more callbacks in Avalanche for every update step, enabling you to add arbitrary plugins after any step of the update process for more flexible and faster idea development. However, I wouldn't expect such a difference for a subset of the dataset. Did you make sure to run both experiments on the same device?

**Comment by 3koozy** [comment]
Hi there @HamedHemati , thank you for the prompt support :).
I tried to make a reproducible code of the issue and to no surprise, I think I figured out where the bottleneck ^^".
it turns out using evaluator plugins only adds a couple of seconds to the loop, but the real issue was when I tried to use logging on a mounted Google drive folder ^^". the epoch period is almost 4x the original and takes around 20 sec/iter regardless of the type of GPU I use or dataset >_<.

I would love to find a solution to this as my standard training procedure takes many hours and I need to save checkpoints and records in remote storage and not risk being removed due to Colab instance disconnect.

Regards.

**Comment by AndreaCossu** [comment]
Hi @3koozy , the mounted Google drive folder may definitely be the answer if you choose a different logging frequency in Avalanche with respect to your original code. For example, if you save to disk every iteration with Avalanche while you save only after each epoch with your code that produces many more disk access operations. 
One solution may be to use only coarse grained metrics (saving stuff after each epoch or after each experience only) or to write a custom logger which saves on a local disk the metric and pushes to the remote drive only later on.

**Comment by 3koozy** [comment]
Hi @AndreaCossu , thanks for the tip :).
I was really thinking about your second solution (save locally and push remotely later) as the answer here.
But I am interested in how can I change the logging frequency (only after epoch/experience) using avalanche built-in methods ?
or do I have to write my own?

**Comment by 3koozy** [comment]
Also, I wanted to add that after removing the per MB option for all the metrics, Diskusage metrics and timing metrics performance substantially improved.

**Comment by AntonioCarta** [comment]
It makes sense because you are logging much more values with MB metrics.
